### ISSUE TITLE
I need to be able to specify the target host from the hosting webpage

### ISSUE THREAD

**Issue by pm100** [swagger-api/swagger-ui#1402]
This is a feature request. I would code it myself but cant find the right way to do it nicely.

At the time I build the swagger.json file I dont know the target web server. I only know it at runtime when the swaggerUI object is started. I need to be able to do something like

      window.swaggerUi = new SwaggerUi({
        url: url,
        host:host,
......

swagger-client/client.js seems to be the right place but its too far away from the swaggerUI object (I could just shove something into the global name space but thats pretty hacky)

**Comment by ebuildy** [comment]
+1 This is so important, cannot understand how guys can work like this. I have edited the swaggerClient, quick & dirty fix ;-(

**Comment by ebuildy** [comment]
Ok, here a very clean way to fix it without edit SwaggerClient, the solution is to load JSON on your own and inject it via **spec** option:

````
$.ajax({
            url: url,
            type: 'GET',
            dataType: 'json',
            contentType: 'application/json',
            ifModified: true,
            cache: false,
            complete: function (xhr, status)
            {
                var spec = JSON.parse(xhr.responseText);

                spec.host = window.location.host;

                window.swaggerUi = new SwaggerUi({
                    url : url, // This will make Swagger happy 
                    dom_id: "swagger-ui-container",
                    spec: spec,
...
                });

                window.swaggerUi.load();
            }
        });
````

Took me 1 hour to figure out set URL is required as well (known issue).

**Comment by fehguy** [comment]
see https://github.com/swagger-api/swagger-js/issues/532

**Comment by webron** [comment]
Fixed in 2.1.1.
